frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,110 @@
+      return this.memberCount() >= 3
     },
-    remove (index) {
-      this.members.splice(index, 1)
+    ...mapState([
+      'currentGroupId',
+      'loggedIn'
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'memberCount'
+    ])
+  },
+  methods: {
+    updateInvitees (payload) {
+      this.form.errorMsg = null
+      Object.assign(this.form, payload.data)
     },
     async submit () {
       try {
-        this.errorMsg = null
-        // TODO: as members are successfully invited display in a
-        // seperate invitees grid and add them to some validation for duplicate invites
-        for (let member of this.members) {
-          // We need to have the latest mailbox attribute for the user
-          const mailbox = await backend.latestHash(member.state.attributes.mailbox)
-          const sentDate = new Date().toString()
+        this.form.errorMsg = null
+        const groupId = this.currentGroupId
+        const groupName = this.currentGroupState.groupName
+
+        for (let member of this.form.invitees) {
+          let mailbox = member.state.attributes.mailbox
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('proposals/invite/mailRecord/create', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('proposals/invite/groupRecord/create', inviteToMailbox.toHash(), memberName, groupId)
+
+          if (this.isProposal) {
+            let latest = await backend.latestHash(groupId)
+            let proposal = new Events.HashableGroupProposal({
+              proposal: template(
+                L('This is a Vote for {name} to become a member of {group}'),
+                { name: memberName, group: groupName }
```

Comment:
OK, so I'm thinking about what will happen in practice here, and I'm guessing that this is probably not the correct behavior, because this is what I think will happen:

- When the proposal is created, the string `'This is a Vote for {name} to become a member of {group}'` is passed through the localizer function, `L`, and so it will return, localized in the native language of whoever made the proposal
- It will then pass through the `template` function, which, I am guessing, will replace the `{name}` and `{group}` parts with the corresponding values
- This, now post-localized string, will be stored in the contract chain
- Then, a user who has a different native language set (say, Spanish), will then have this English string displayed in their UI, not localized.

So the data that's saved needs to be something else. Admittedly this is not your fault, but how it was designed before by Victor, and it is also my fault for not catching it back then when he submitted the pull request.

But good thing I caught it now! 😄 

(And please verify that my understanding is correct.)